server/src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -530,7 +463,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (lastAllocatedNodeId != null) {
             builder.field("last_node", lastAllocatedNodeId);
         }
-        String details = getDetails();
+        String details = details();
         if (details != null) {
             builder.field("details", details);
         }
```

Comment:
Current equals and hashCode implementation ignores `unassignedTimeNanos` for some reason.
I do not think there is a specific reason for that. I suggest to open a followup pr to try to remove custom implementation in favor of the default one.